src/server/run-websocket-server.ts

```diff
@@ -2,68 +2,86 @@ import * as WebSocket from "ws";
 import * as Knex from "knex";
 import { EventEmitter } from "events";
 import { augurEmitter } from "../events";
-import { JsonRpcRequest } from "../types";
+import { JsonRpcRequest, WebSocketConfigs } from "../types";
 import { isJsonRpcRequest } from "./is-json-rpc-request";
 import { dispatchJsonRpcRequest } from "./dispatch-json-rpc-request";
 import { makeJsonRpcResponse } from "./make-json-rpc-response";
 import { makeJsonRpcError, JsonRpcErrorCode } from "./make-json-rpc-error";
 import { Subscriptions } from "./subscriptions";
+import * as fs from "fs";
+import * as https from "https";
 
-export function runWebsocketServer(db: Knex, port: number): WebSocket.Server {
-  console.log("Starting websocket server on port", port);
-  const websocketServer: WebSocket.Server = new WebSocket.Server({ port });
-  websocketServer.on("connection", (websocket: WebSocket): void => {
-    const subscriptions = new Subscriptions(augurEmitter);
+export function runWebsocketServer(db: Knex, webSocketConfigs: WebSocketConfigs): Array<WebSocket.Server> {
 
-    websocket.on("message", (data: WebSocket.Data): void => {
-      let message: any;
-      try {
-        message = JSON.parse(data as string);
-        if (!isJsonRpcRequest(message)) return console.error("bad json rpc message received:", message);
-      } catch (exc) {
-        return websocket.send(makeJsonRpcError("-1", JsonRpcErrorCode.ParseError, "Bad JSON RPC Message Received", { originalText: data as string }));
-      }
+  const websocketServers: Array<WebSocket.Server> = [];
 
-      try {
-        if (message.method === "subscribe") {
-          const eventName: string = message.params.shift();
+  if ( webSocketConfigs.wss != null ) {
+    console.log("Starting websocket secure server on port", webSocketConfigs.wss.port);
+    const server = https.createServer({
+      cert: fs.readFileSync(webSocketConfigs.wss.certificateFile),
```

Comment:
Reasonable way to read a file here? Mostly what i've read says that it's OK to do sync things during setup when you can't proceed until it is done

Reply:
yeah thats fine, can't start up without it if its specified.